src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -712,6 +715,18 @@ class Peer(
         peerConnection?.send(message)
     }
 
+    suspend fun requestAddress(languageSubtag: String): String {
```

Comment:
This will suspend forever if there is no channel and the peer doesn't answer, we should at least document it to make sure the app correctly handles this? I think it's important to make sure we seamlessly handle the case where the peer doesn't answer, because that will allow us to update the protocol in the future and simply stop responding to those requests once we want to make replace them with a new protocol?

Reply:
Current implementation on Phoenix uses a 5s timeout, we can work with that or move the timeout into lightning-kmp.

Reply:
Documented the behavior in 146ffd4b.

Reply:
> Current implementation on Phoenix uses a 5s timeout, we can work with that or move the timeout into lightning-kmp.

I think it makes sense to have a timeout on the Phoenix side here. This way, if we make this message obsolete at some point, old versions of the app will just go through the timeout failure path.